R/download_px_file.R

```diff
@@ -0,0 +1,69 @@
+#' Download px cube file for px id into a local directory
+#'
+#' @import httr
+#' @importFrom utils download.file
+#'
+#' @param download_url url to download from
+#' @param destfile file path to download the file to
+#'
+#' @return downloaded_file
+#' @noRd
+#'
+#' @examples download_file_from_url('https://www.pxweb.bfs.admin.ch/DownloadFile.aspx?file=px-x-0102020207_102', '/tmp/')
+download_file_from_url <- function(download_url, destfile) {
+  tryCatch(
+    {
+      file <- GET(download_url)
+      file_size <- headers(file)$`content-length`
+      rc <- download.file(download_url, destfile=destfile)
+      if (rc != 0) {
+        stop("File could not be downloaded")
+      }
+      if (file.size(destfile) != file_size) {
+        stop("Downloaded file has not the expected size")
```

Comment:
```suggestion
        file.remove(destfile)
        stop("Download succeeded BUT the file did not have the expected size. The file has been removed.")
```

I would go for removing the file downloaded if the dimensions do not match